cpp/custrings/strings/attrs.cu

```diff
@@ -437,3 +438,43 @@ unsigned int NVStrings::is_empty( bool* results, bool todevice )
     return (unsigned int)matches;
 }
 
+//
+// s = ["a","xyz", "éee"]
+// s.code_points(results)
+// results is [   97   120   121   122 50089   101   101]
+//
+unsigned int NVStrings::code_points( unsigned int* d_results )
+{
+    auto count = size();
+    auto execpol = rmm::exec_policy(0);
+    custring_view_array d_strings = pImpl->getStringsPtr();
+
+    // offsets point to each individual integer range
+    rmm::device_vector<size_t> offsets(count);
+    size_t* d_offsets = offsets.data().get();
+    thrust::transform_inclusive_scan(execpol->on(0),
+        thrust::make_counting_iterator<size_t>(0),
+        thrust::make_counting_iterator<size_t>(count),
+        d_offsets,
+        [d_strings] __device__(size_t idx){
+            custring_view* d_str = d_strings[idx];
+            size_t length = 0;
+            if( d_str )
+                length = d_str->chars_count();
+            return length;
+        },
+        thrust::plus<unsigned int>());
+
+    // now set the ranges from each strings' character values
+    thrust::for_each_n(execpol->on(0), thrust::make_counting_iterator<unsigned int>(0), count,
+        [d_strings, d_offsets, d_results] __device__(unsigned int idx){
+            custring_view* d_str = d_strings[idx];
+            if( !d_str )
+                return;
+            auto result = d_results + (idx ? d_offsets[idx-1] : 0);
```

Comment:
If you replace the inclusive scan with an exclusive scan, then this line just becomes 
```suggestion
            auto result = d_results + d_offsets[idx-1];
```

Reply:
I think you meant `result = d_results + d_offsets[idx];`. Yes that works.